src/dmap_import/util_rds.py

```diff
@@ -0,0 +1,306 @@
+import os
+import platform
+import urllib.parse as urlparse
+from typing import Any, Dict, List, Tuple, Union
+
+import boto3
+import sqlalchemy as sa
+from sqlalchemy.orm import sessionmaker
+from alembic.config import Config
+from alembic import command
+
+from dmap_import.util_logging import ProcessLogger
+
+
+def running_in_docker() -> bool:
+    """
+    return True if running inside of a docker container, else False
+    """
+    path = "/proc/self/cgroup"
+    return (
+        os.path.exists("/.dockerenv")
+        or os.path.isfile(path)
+        and any("docker" in line for line in open(path, encoding="UTF-8"))
+    )
+
+
+def running_in_aws() -> bool:
+    """
+    return True if running on aws, else False
+    """
+    return bool(os.getenv("AWS_DEFAULT_REGION"))
+
+
+def get_db_password() -> str:
+    """
+    function to provide rds password
+
+    used to refresh auth token, if required
+    """
+    db_password = os.environ.get("DB_PASSWORD", None)
+    db_host = os.environ.get("DB_HOST")
+    db_port = os.environ.get("DB_PORT")
+    db_user = os.environ.get("DB_USER")
+    db_region = os.environ.get("DB_REGION", None)
+
+    if db_password is None:
+        # generate aws db auth token if in rds
+        client = boto3.client("rds")
+        return client.generate_db_auth_token(
+            DBHostname=db_host,
+            Port=db_port,
+            DBUsername=db_user,
+            Region=db_region,
+        )
+
+    return db_password
+
+
+def postgres_event_update_db_password(
+    _: sa.engine.interfaces.Dialect,
+    __: Any,
+    ___: Tuple[Any, ...],
+    cparams: Dict[str, Any],
+) -> None:
+    """
+    update database passord on every new connection attempt
+    this will refresh db auth token passwords
+    """
+    process_logger = ProcessLogger("password_refresh")
+    process_logger.log_start()
+    cparams["password"] = get_db_password()
+    process_logger.log_complete()
+
+
+def get_local_engine(
+    echo: bool = False,
+) -> sa.future.engine.Engine:
+    """
+    Get an SQL Alchemy engine that connects to a locally Postgres RDS stood up
+    via docker using env variables
+    """
```

Comment:
Most of this file also copied over from LAMP.

Seems to work pretty will with AWS RDS.